Code/GraphMol/RGroupDecomposition/RGroupDecomp.h

```diff
@@ -15,6 +15,10 @@
 #include <GraphMol/Substruct/SubstructMatch.h>
 #include <chrono>
 
+#ifdef RDK_BUILD_CPP_TESTS
```

Comment:
Does this really need to be in the main header file if it's only used in the tests?
Can you just include the declaration in the tests themselves?

Reply:
This is a forward declaration for (Line 130)
```cpp
#ifdef RDK_BUILD_CPP_TESTS
  friend void ::testCoresLabelledProperly();
#endif
```

I guess I don't know how to move the friend declaration to the test class and it won't compile without the forward declaration.

I wanted to be able to test the internal structure of the private RGroup data structures- it was hard to catch labeling errors otherwise.

The friend declaration seems like an anti-pattern, but I didn't want to include creating a public API in this PR.

Reply:
Ah right. 
I have an alternate proposal for solving this - creating a new test file for internals and doing `#define private public` before including RGroupDecomp.h in that file - that doesn't require this modification or changing the global CMakeLists.txt.
Will do a PR against your branch with that suggestion.

Reply:
Good idea.